packages/plugin-metrics/test/unit/spec/batcher.js

```diff
@@ -0,0 +1,168 @@
+/**!
+ *
+ * Copyright (c) 2015 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {assert} from '@ciscospark/test-helper-chai';
+import lolex from 'lolex';
+import Metrics, {config} from '../..';
+import MockSpark from '@ciscospark/test-helper-mock-spark';
+import sinon from '@ciscospark/test-helper-sinon';
+import {SparkHttpError} from '@ciscospark/spark-core';
+
+function promiseTick(count) {
+  let promise = Promise.resolve();
+  while (count > 1) {
+    promise = promise.then(() => {
+      return promiseTick(1);
```

Comment:
Why the recursion? Doesn't promiseTick just return Promise.resolve() if count === 1?Sorry if I'm missing something